admin/cleanup.py

```diff
@@ -0,0 +1,379 @@
+            )
+        )
+    return file_path
+
+
+def _yaml_configuration_path_option(option_name, option_value):
+    yaml_path = _existing_file_path_option(option_name, option_value)
+    try:
+        configuration = yaml.safe_load(yaml_path.open())
+    except ParserError as e:
+        raise UsageError(
+            u"Problem with --{}. "
+            u"Unable to parse YAML from {}. "
+            u"Error message: {}.".format(
+                option_name, yaml_path.path, unicode(e)
+            )
+        )
+    return configuration
+
+
+class CleanupCloudResourcesOptions(Options):
+    """
+    """
+    optFlags = [
+        ["dry-run", None,
+         "Just print the calculated actions. Don't delete anything."]
+    ]
+
+    optParameters = [
+        [u"config-file", None, None,
+         u"The config file containing cloud credentials.\n",
+         lambda option_value: _yaml_configuration_path_option(
+             u"config-file", option_value
+         )],
+        [u"volume-lag", None, timedelta(minutes=30),
+         u"The oldest in minutes a volume may be "
+         u"without being considered for deletion.\n",
+         lambda option_value: timedelta(minutes=int(option_value))],
+        [u"marker", None, MARKER,
+         u"The marker which is used "
+         u"as the ``node`` in the acceptance test cluster UUID.",
+         lambda option_value: int(option_value, base=16)]
+    ]
+
+    def postOptions(self):
+        self["dry-run"] = bool(self["dry-run"])
+        if self["config-file"] is None:
+            raise UsageError("Missing --config-file option.")
+
+
+def cleanup_cloud_resources_main(reactor, args, base_path, top_level):
+    options = CleanupCloudResourcesOptions()
+
+    try:
+        options.parseOptions(args)
+    except UsageError as e:
+        sys.stderr.write(
+            u"{}\n"
+            u"Usage Error: {}: {}\n".format(
+                unicode(options), base_path.basename(), e
+            ).encode('utf-8')
+        )
+        raise SystemExit(1)
+
+    cleaner = CleanVolumes(
+        lag=options["volume-lag"],
+        marker=options["marker"],
+    )
+
+    actions = cleaner.start(
+        config=options["config-file"],
+        dry_run=options["dry-run"],
+    )
+    d = succeed(actions)
+    d.addCallback(print_result)
+    return d
+
+
+def print_result(actions):
+    """
+    If volumes are destroyed, the operation is considered to have failed.
+    The test suite should have cleaned those volumes up.  This is an
+    unfortunate time to be reporting the problem but it's better than never
+    reporting it.
+    """
+    sys.stdout.write(
+        _serialize_volume_actions(actions).encode('utf-8') + b'\n'
+    )
+    if len(actions.destroy) > 0:
+        # We fail if we destroyed any volumes because that means that
+        # something is leaking volumes.
+        raise SystemExit(1)
+
+
+def _serialize_volume_actions(actions):
+    serializable = {}
+    for kind in ('destroy', 'keep'):
```

Comment:
:question: should we print out the keep? I guess it's nice if you want to be like, "why isn't my volume being deleted", but it's kind of noisy in general. (running this in us-west-2 gets 1517 lines of output). That said I can always pipe into less or use jq, so maybe better as is.